### PR TITLE
Vizapp!

### PR DESCRIPTION
**Pull request by EthanMarx**
Adds back vizapp with the addition of spectrogram plotting.

Keeps old plotting code in `plots/legacy` so that the SV task won't break. Down the line we can get rid of this and use the vizapp plotting code directly (didn't do this now b/c required going down a refactoring rabit hole I didn't wan't to undertake).

### CODE REVIEW

**Comment by EthanMarx** [comment]
Also, sanity checked the SV `luigi` task and ran successfully.

**Comment by EthanMarx** [comment]
@wbenoit26 No rush on this but I think this is more or less ready to be reviewed

**Comment by wbenoit26** [comment]
Oh, awesome, I'll give it a look this afternoon

File: projects/plots/plots/vizapp/infer/analyzer.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            - int(self.sample_rate * self.fduration)
+        )
+        return self.times[start:]
+
+    def find_strain(self, time: float, shifts: Sequence[float]):
+        # find strain file corresponding to requested time
+        fname, t0, duration = get_strain_fname(self.strain_dir, time)
+        # find indices of data needed for inference
+        times = np.arange(t0, t0 + duration, 1 / self.sample_rate)
+        start, stop = get_indices(
```
Comment (wbenoit26) [inline projects/plots/plots/vizapp/infer/analyzer.py:132]: Won't `stop` be an index passed the end of the time array?

**Review by wbenoit26 — COMMENTED** [review]
I had a few questions/comments, but generally I think this looks good. I admit that I didn't look too closely at some of the plot formatting code, but if it all looks good then I imagine that it's working

File: projects/plots/plots/vizapp/app.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+        # load in the model weights
+        self.model = self.load_model()
+
+        # instantiate object for managing
+        # data loading and application of vetos
+        self.data_manager = DataManager(results_dir, data_dir, ifos)
+
+        # initialize all our pages and their constituent plots
+        self.pages: list["Page"] = []
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/plots/plots/vizapp/app.py:80]: Just type hinting a variable to please my IDE

File: projects/plots/plots/vizapp/data.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+class DataManager:
+    """
+    Class for managing data, including applying vetos
+    """
+
+    def __init__(self, results_dir: Path, data_dir: Path, ifos: list[str]):
+        self.logger = logging.getLogger("vizapp")
+        self.ifos = ifos
+        # load results and data from the run we're visualizing
+        infer_dir = results_dir / "infer" / "1year"
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/plots/plots/vizapp/data.py:40]: ah yeah that was from my debugging

File: projects/plots/plots/vizapp/infer/analyzer.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    @property
+    def times(self):
+        """
+        Returns the time values relative to event time
+        """
+        start = (
+            self.psd_length
+            + self.kernel_length
+            + (self.fduration / 2)
+            + self.padding
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/plots/plots/vizapp/infer/analyzer.py:109]: Padding is amount of seconds after the coalescence time exits the kernel that we query

File: projects/plots/plots/vizapp/infer/analyzer.py (hunk 1 above)
Comment (EthanMarx) [reply projects/plots/plots/vizapp/infer/analyzer.py:132]: Not sure what you mean

Comment (wbenoit26) [reply projects/plots/plots/vizapp/infer/analyzer.py:132]: Not sure what I meant by this either, to be honest. I think I misread something the first time I reviewed.

When is `time` relative to the coalescence point?

**Comment by EthanMarx** [comment]
@wbenoit26 addressed your concerns

File: projects/plots/plots/vizapp/pages/summary/sv.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+class SensitiveVolumePlot:
+    def __init__(
+        self,
+        background: EventSet,
+        foreground: RecoveredInjectionSet,
+        rejected_params: InjectionParameterSet,
+        mass_combos: List[tuple],
+        source_prior: PriorDict,
+        dt: Optional[float] = None,
+        max_far: float = 100 / SECONDS_PER_MONTH,
```
Comment (wbenoit26) [inline projects/plots/plots/vizapp/pages/summary/sv.py:36]: It needs to be `SECONDS_PER_YEAR` here as well

Comment (EthanMarx) [reply projects/plots/plots/vizapp/pages/summary/sv.py:36]: Does it? Im just defining the max_far as 100 per month

Comment (wbenoit26) [reply projects/plots/plots/vizapp/pages/summary/sv.py:36]: Yeah, "need" was the wrong word. I was thinking that 1200 per year seemed like kind of a high FAR to go to. But that's a minor point

**Review by wbenoit26 — APPROVED** [review]

File: projects/plots/plots/vizapp/pages/summary/sv.py (hunk 5 above)
Comment (EthanMarx) [reply projects/plots/plots/vizapp/pages/summary/sv.py:36]: got it, yeah thats probably right

**Pull request merged**